src/group/managed_group/mod.rs

```diff
@@ -517,23 +517,19 @@ impl<'a> ManagedGroup<'a> {
         if !self.active {
             return Err(ManagedGroupError::UseAfterEviction);
         }
-        // Use provided KeyPackageBundle or create a new one
-        let tree = self.group.tree();
-        let key_package_bundle = match key_package_bundle_option {
-            Some(kpb) => kpb,
-            None => {
-                let existing_key_package = tree.own_key_package();
-                KeyPackageBundle::from_rekeyed_key_package(existing_key_package)
-            }
-        };
-        drop(tree);
 
-        // Create UpdateProposal
-        let mut plaintext_messages = vec![self.group.create_update_proposal(
-            &self.aad,
-            &self.credential_bundle,
-            key_package_bundle.key_package().clone(),
-        )];
+        // If a KeyPackageBundle was provided, create an UpdateProposal
+        let mut plaintext_messages = if let Some(key_package_bundle) = key_package_bundle_option {
```

Comment:
What are you changing here in terms of functionality? Why wouldn't you just create an update proposal with a re-keyed key package?

Reply:
The `UpdateProposal` was unnecessary, because we will overwrite it anyway with a full Commit. This way we save some cost on crypto primitives and enforce a full Commit instead.